From 103a8d30ba9030f3df66debc0baa94180f154a7e Mon Sep 17 00:00:00 2001
From: Bin Yang <bin.y.yang@intel.com>
Date: Sun, 2 Jun 2019 20:18:47 +0800
Subject: [PATCH 2/2] emulate kernel cmdline to deal with parameters

Change-Id: Iab7e2d936546cecd851d504ca643c63b8a5d52cb
Tracked-On:
Signed-off-by: Bin Yang <bin.y.yang@intel.com>
(cherry picked from commit c0ce6b09ad0a76afaf3d5e137d202171cebc54ea)
---
 init/init.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index fc58eeabb..dfff85fe7 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -384,6 +384,9 @@ static void export_kernel_boot_props() {
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
         { "ro.boot.hardware",   "ro.hardware",   "unknown", },
         { "ro.boot.revision",   "ro.revision",   "0", },
+        { "ro.boot.container.id",       "ro.container.id",       "unknown",  },
+        { "ro.boot.container.memnodes", "ro.container.memnodes", "unknown",  },
+        { "ro.boot.container.cpus",     "ro.container.cpus",     "unknown",  },
     };
     for (size_t i = 0; i < arraysize(prop_map); i++) {
         std::string value = GetProperty(prop_map[i].src_prop, "");
@@ -580,6 +583,7 @@ int main(int argc, char** argv) {
         mount("devpts", "/dev/pts", "devpts", 0, NULL);
         #define MAKE_STR(x) __STRING(x)
         mount("proc", "/proc", "proc", 0, "hidepid=2,gid=" MAKE_STR(AID_READPROC));
+        mount("/cmdline", "/proc/cmdline", NULL, MS_BIND|MS_RDONLY, NULL);
         // Don't expose the raw commandline to unprivileged processes.
         chmod("/proc/cmdline", 0440);
         gid_t groups[] = { AID_READPROC };
-- 
2.21.0

